nsxt/policy_utils.go

```diff
@@ -410,7 +411,33 @@ func validateImportPolicyPath(policyPath string) error {
 	return nil
 }
 
-// This importer function accepts policy path and resource ID
+func isValidResourceID(id string) bool {
+	if isSpaceString(id) {
+		return false
+	}
+	if len(id) > 1024 {
+		return false
+	}
+	if strings.Contains(id, "/") {
+		return false
+	}
+	return true
+}
+
+// This importer function accepts policy path only as import ID
+func getFriendlyPolicyPathResourceImporter(pathExample string) func(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
```

Comment:
I have no problem with the approach you followed, but why did you choose to create a function that returns an importer rather than defining directly the importer function?

Reply:
Clarified live with Anna.